mercury_ucx_parse_addr_change.patch

```diff
@@ -0,0 +1,129 @@
+         "ucp_worker_create() failed (%s)", ucs_status_string(status));
+ 
+     /* Print worker info */
++    /*
+     NA_LOG_SUBSYS_DEBUG_FUNC(ctx,
+         ucp_worker_print_info(worker, hg_log_get_stream_debug()),
+         "Worker info");
+-
++    */
+     /* Check thread mode */
+     status = ucp_worker_query(worker, &worker_attrs);
+     NA_CHECK_SUBSYS_ERROR(cls, status != UCS_OK, error, ret, NA_PROTOCOL_ERROR,
+@@ -1958,45 +1961,50 @@ static na_return_t
+ na_ucx_parse_hostname_info(const char *hostname_info, const char *subnet_info,
+     char **net_device_p, struct sockaddr_storage **sockaddr_p)
+ {
+-    char **ifa_name_p = NULL;
+     char *hostname = NULL;
++    //char host[64];
++    char *device_delim = NULL;
+     unsigned int port = 0;
+     na_return_t ret = NA_SUCCESS;
+-
+     /* Set hostname (use default interface name if no hostname was passed) */
+     if (hostname_info) {
+-        hostname = strdup(hostname_info);
+-        NA_CHECK_SUBSYS_ERROR(cls, hostname == NULL, done, ret, NA_NOMEM,
+-            "strdup() of hostname failed");
++        device_delim = strstr(hostname_info, "/");
++	if (device_delim) {
++	    char *addr_str = NULL;
++            strtok_r(hostname_info, "/", &addr_str);
++            hostname = strdup(addr_str);
++            NA_CHECK_SUBSYS_ERROR(cls, hostname == NULL, done, ret, NA_NOMEM,
++                          "strdup() of hostname failed");
++            /* put NULL in hostname_info string after device name */
++	    *(++device_delim) = '\0';
++	    *net_device_p = hostname_info;
++	} else {
++            hostname = strdup(hostname_info);
++            NA_CHECK_SUBSYS_ERROR(cls, hostname == NULL, done, ret, NA_NOMEM,
++                          "strdup() of hostname failed");
++	}
+ 
+         /* TODO add support for IPv6 address parsing */
+ 
+-        /* Extract hostname : port */
++        /* Extract hostname / port */
+         if (strstr(hostname, ":")) {
+             char *port_str = NULL;
+             strtok_r(hostname, ":", &port_str);
+             port = (unsigned int) strtoul(port_str, NULL, 10);
+         }
+-
+-        /* Extract net_device if explicitly listed with '/' before IP */
+-        if (strstr(hostname, "/")) {
+-            char *host_str = NULL;
+-            strtok_r(hostname, "/", &host_str);
+-
+-            *net_device_p = hostname;
+-            hostname = strdup(host_str);
+-            NA_CHECK_SUBSYS_ERROR(cls, hostname == NULL, done, ret, NA_NOMEM,
+-                "strdup() of hostname failed");
+-        } else
+-            ifa_name_p = net_device_p;
+     }
+ 
+     /* TODO add support for IPv6 wildcards */
+ 
+     if (hostname && strcmp(hostname, "0.0.0.0") != 0) {
+-        /* Try to get matching IP/device */
+-        ret = na_ip_check_interface(hostname, port, ifa_name_p, sockaddr_p);
+-        NA_CHECK_SUBSYS_NA_ERROR(cls, done, ret, "Could not check interfaces");
++	struct sockaddr_storage *ss_addr = calloc(1, sizeof(*ss_addr));
```

Comment:
Need to free this?

Reply:
It is freed in the calling function: na_ucx_initialize().

Reply:
There a few commented lines of code included in the patch.  These lines are retained as comments to enable integration of the code changes into the Mercury code base.